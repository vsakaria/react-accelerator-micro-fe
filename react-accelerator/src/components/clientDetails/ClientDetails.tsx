import React from "react";
import { IClient } from "../IClient";
import { withErrorBoundary } from "../../utils/components/ErrorBoundaryHoC";
import styles from "./styles/ClientDetails.module.css";

export const ClientDetails: React.FC<IClient> = ({
  id,
  name,
  status,
  adminType,
  passwordExpiryPeriod,
  activeFrom
}) => {
  return (
    <section className={styles.clientDetailContainer}>
      <h4 className="primary-color">Client details</h4>
      <table className={styles.clientDetailTable}>
        <tbody>
          <tr>
            <td>Client id</td>
            <td>{id}</td>
          </tr>
          <tr>
            <td>Client name</td>
            <td>{name}</td>
          </tr>
          <tr>
            <td>Client status</td>
            <td>{status}</td>
          </tr>
          <tr>
            <td>Administration type</td>
            <td>{adminType}</td>
          </tr>
          <tr>
            <td>Password expiry period</td>
            <td>{`${passwordExpiryPeriod} ${
              passwordExpiryPeriod > 1 ? "days" : "day"
            }`}</td>
          </tr>
          <tr>
            <td>Active form</td>
            <td>{activeFrom}</td>
          </tr>
        </tbody>
      </table>
    </section>
  );
};

export default withErrorBoundary(ClientDetails);
