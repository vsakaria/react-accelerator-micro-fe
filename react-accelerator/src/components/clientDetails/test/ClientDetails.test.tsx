import React from "react";
import { ClientDetails } from "../ClientDetails";
import { shallowSnapShot } from "../../../utils/testing/testing";
import { shallow } from "enzyme";

describe("ClientSummary", () => {
  let props = {
    id: 49084369,
    name: "AutoCBS1",
    status: "Active",
    adminType: "Single Admin",
    passwordExpiryPeriod: 90,
    activeFrom: "24/05/2019"
  };

  it("should display the title", () => {
    const clientSummarySS = shallowSnapShot(<ClientDetails {...props} />);
    expect(clientSummarySS).toMatchSnapshot();
  });

  it("should display the title", () => {
    props.passwordExpiryPeriod = 1;

    const clientSummarySS = shallow(<ClientDetails {...props} />);
    const dayNodeText = clientSummarySS
      .find("td")
      .at(9)
      .text();

    expect(dayNodeText).toEqual("1 day");
  });
});
