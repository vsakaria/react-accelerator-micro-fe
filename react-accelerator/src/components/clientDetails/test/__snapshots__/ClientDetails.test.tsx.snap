// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientSummary should display the title 1`] = `
<section
  className="clientDetailContainer"
>
  <h4
    className="primary-color"
  >
    Client details
  </h4>
  <table
    className="clientDetailTable"
  >
    <tbody>
      <tr>
        <td>
          Client id
        </td>
        <td>
          49084369
        </td>
      </tr>
      <tr>
        <td>
          Client name
        </td>
        <td>
          AutoCBS1
        </td>
      </tr>
      <tr>
        <td>
          Client status
        </td>
        <td>
          Active
        </td>
      </tr>
      <tr>
        <td>
          Administration type
        </td>
        <td>
          Single Admin
        </td>
      </tr>
      <tr>
        <td>
          Password expiry period
        </td>
        <td>
          90 days
        </td>
      </tr>
      <tr>
        <td>
          Active form
        </td>
        <td>
          24/05/2019
        </td>
      </tr>
    </tbody>
  </table>
</section>
`;
